c7n/resources/__init__.py

```diff
@@ -76,3 +76,7 @@ def load_resources():
     # Load external plugins (private sdks etc)
     from c7n.manager import resources
     resources.load_plugins()
+
+    for _, klass in resources.items():
```

Comment:
rather than run events per Klass, action/filter registry. run once for the resource type registry.